datafusion/proto/tests/cases/roundtrip_physical_plan.rs

```diff
@@ -272,31 +269,34 @@ fn roundtrip_window() -> Result<()> {
     let field_b = Field::new("b", DataType::Int64, false);
     let schema = Arc::new(Schema::new(vec![field_a, field_b]));
 
-    let window_frame = WindowFrame::new_bounds(
+    let _window_frame = WindowFrame::new_bounds(
         datafusion_expr::WindowFrameUnits::Range,
         WindowFrameBound::Preceding(ScalarValue::Int64(None)),
         WindowFrameBound::CurrentRow,
     );
 
-    let builtin_window_expr = Arc::new(BuiltInWindowExpr::new(
-        Arc::new(NthValue::first(
-            "FIRST_VALUE(a) PARTITION BY [b] ORDER BY [a ASC NULLS LAST] RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW",
-            col("a", &schema)?,
-            DataType::Int64,
-            false,
-        )),
-        &[col("b", &schema)?],
-        &LexOrdering{
-            inner: vec![PhysicalSortExpr {
-                expr: col("a", &schema)?,
-                options: SortOptions {
-                    descending: false,
-                    nulls_first: false,
-                },
-            }]
-        },
-        Arc::new(window_frame),
-    ));
+    // let udwf_window_expr = Expr::WindowFunction(WindowFunction::new(
+    //     WindowFunctionDefinition::WindowUDF(first_value_udwf()),
+
+    // let builtin_window_expr = Arc::new(BuiltInWindowExpr::new(
+    //     Arc::new(NthValue::first(
+    //         "FIRST_VALUE(a) PARTITION BY [b] ORDER BY [a ASC NULLS LAST] RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW",
+    //         col("a", &schema)?,
+    //         DataType::Int64,
+    //         false,
+    //     )),
+    //     &[col("b", &schema)?],
+    //     &LexOrdering{
+    //         inner: vec![PhysicalSortExpr {
```

Comment:
I could not make this expression work with the first_value_udwf? I looked to the similar things but still could not build the required expression. Can I get some help please @jcsherin @berkaysynnada ?

Reply:
I've sent a commit, I believe it will unblock you. Perhaps we can even remove this expression since `SlidingAggregateWindowExpr` has already been added in the test (and `PlainAggregateWindowExpr` has also), but keep it for now.

Reply:
That one fails with: 

```
to proto: NotImplemented("WindowExpr not supported: BuiltInWindowExpr { expr: WindowUDFExpr { fun: WindowUDF { inner: NthValue { signature: Signature { type_signature: OneOf([Any(0), Any(1), Any(2)]), volatility: Immutable }, kind: First } }, args: [Column { name: \"a\", index: 0 }], name: \"FIRST_VALUE(a) PARTITION BY [b] ORDER BY [a ASC NULLS LAST] RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW\", input_types: [Int64], is_reversed: false, ignore_nulls: false }, partition_by: [Column { name: \"b\", index: 1 }], order_by: LexOrdering { inner: [PhysicalSortExpr { expr: Column { name: \"a\", index: 0 }, options: SortOptions { descending: false, nulls_first: false } }] }, window_frame: WindowFrame { units: Range, start_bound: Preceding(Int64(NULL)), end_bound: CurrentRow, is_causal: false } }")
```

Checking if I can fix it

Reply:
This does not work because we can't convert to proto. The issue is that we have the following definition:
```
/// Nested message and enum types in `PhysicalWindowExprNode`.
pub mod physical_window_expr_node {
    #[derive(Clone, PartialEq, ::prost::Oneof)]
    pub enum WindowFunction {
        #[prost(enumeration = "super::BuiltInWindowFunction", tag = "2")]
        BuiltInFunction(i32),
        #[prost(string, tag = "3")]
        UserDefinedAggrFunction(::prost::alloc::string::String),
    }
}
```


It can be UDAF or BuiltinFunction. This deletion https://github.com/apache/datafusion/pull/13201/files#r1836542435, causes the issue. What should I do here? I think deleting BuiltinWindowExpr was ok but there is nothing to replace here AFAIS

Reply:
For this to work we need to add serialization for user-defined window functions. 

I think we can remove `builtin_window_expr` from the round trip test
(but keep the `PlainAggregateWindowExpr` and `SlidingAggregateWindowExpr` tests intact).

Reply:
Yes, that's what I've been saying actually. Then I'm deleting builtin part from test